### PR TITLE
Radioactive Teleport Armor

### PR DESCRIPTION
**Pull request by Boredone**
This Pull Request simply adds a new child of reactive armor... a normal reactive teleport armor that calls rad_act on teleport, radiating the teleporter hopefully.

As a note, this is my first PR, my first time semi-seriously coding, and I forgot to get this looked at first, so please feel free to point out stuff I did wrong.

:cl: Boredone
tweak: Due to an experiment gone wrong, the Research Director's Teleport Armor now causes some mild radiation poisoning on teleportation to the wearer.
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
:^( 
10 maybe? :-)

File: code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -182,6 +182,38 @@
 		return 1
 	return 0
 
+/obj/item/clothing/suit/armor/reactive/teleport
```
Comment (neersighted) [inline code/modules/clothing/suits/armor.dm]: This is great and all, but you should remove the code you copypasta'd from the parent type.

Comment (Boredone) [inline code/modules/clothing/suits/armor.dm]: I suppose that makes sense. Though admins might want to be able to spawn in a safe version, and they wouldn't be able to do that by just varediting it.

**Comment by Dorsisdwarf** [comment]
There seems to be a divide between whether this is gutting the RTA and nerfing the replacement child, or just creating a child object.

**Comment by GunHog** [comment]
Please, no. :-1:

File: code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -182,6 +182,38 @@
...
+			turfs += pick(/turf in orange(6, H))
+		var/turf/picked = pick(turfs)
+		if(!isturf(picked))
+			return
+		if(H.buckled)
+			H.buckled.unbuckle_mob()
+		H.forceMove(picked)
+		H.rad_act(15)
+		return 1
+	return 0
```
Comment (phil235) [inline code/modules/clothing/suits/armor.dm]: Don't copypasta the parent proc, you can just do this instead:
```
	. = ..()
	if(.)
		H.rad_act(15)
```

**Comment by GunHog** [comment]
I posted in the wrong PR before, so I am leaving this here instead:
> I am getting sick of all these grudge code PRs, including the memes. One person (Mekhi) overuses something, and now some salty player with Github installed takes it away. It is horrible and it needs to stop.

**Comment by PKPenguin321** [comment]
Said the RD player with github installed who buffs RD at every opportunity (pointless RD hardsuit never 4get)

File: code/modules/clothing/suits/armor.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/modules/clothing/suits/armor.dm]: do what this man says, boredone. it accomplishes everything this copied code does with an extremely small amount of lines.

**Comment by hornygranny** [comment]
just because most people are extremely unrobust doesn't mean we don't have to balance the game properly

File: code/modules/clothing/suits/armor.dm (hunk 2 above)
Comment (Boredone) [inline code/modules/clothing/suits/armor.dm]: Attempting that has the error 'code\modules\clothing\suits\armor.dm:193:error: H.rad_act: undefined var' for some reason.

**Comment by PKPenguin321** [comment]
hg with the double negatives mindgames

**Comment by neersighted** [comment]
The subtype really should be more descriptively named. Ideally the base teleport armor is `reactive/teleport` and this one is `reactive/teleport/harmful` or something.

**Comment by Boredone** [comment]
Well, did some testing, works fine as far as I can tell, main problem is it doesn't warn the user it radiated them for some reason.
For those who didn't check the commits, removed the teleport from the armor/reactive parent, and set the range and radiation on the teleport armor to vars, so admins can mess with them, or make them safe pretty easily if they choose.

File: code/modules/clothing/suits/armor.dm [hunk 3]
```diff
@@ -156,29 +156,37 @@
 	src.item_state = "reactiveoff"
 	..()
 
-/obj/item/clothing/suit/armor/reactive/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance)
+/obj/item/clothing/suit/armor/reactive/teleport
+	name = "reactive teleport armor"
+	desc = "Someone seperated our Research Director from his own head!"
+	hit_reaction_chance = 50
```
Comment (phil235) [inline code/modules/clothing/suits/armor.dm]: already set in /obj/item/clothing/suit/armor/reactive

File: code/modules/clothing/suits/armor.dm [hunk 4]
```diff
@@ -156,29 +156,37 @@
 	src.item_state = "reactiveoff"
 	..()
 
-/obj/item/clothing/suit/armor/reactive/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance)
+/obj/item/clothing/suit/armor/reactive/teleport
+	name = "reactive teleport armor"
+	desc = "Someone seperated our Research Director from his own head!"
```
Comment (phil235) [inline code/modules/clothing/suits/armor.dm]: this should probably have a different description than /obj/item/clothing/suit/armor/reactive

**Comment by Boredone** [comment]
So, unless I missed something I shouldn't be doing codewise, seems fine now cept for the fact it doesn't show the 'Your Clothes Feel Warm' thing when you teleport for some unknown reason.
I should also probobly add a changelog though now that I think about it.

**Comment by tkdrg** [comment]
Did you really intend to replace the RD's suit with this? This is uh kind of a giant nerf.

**Comment by Boredone** [comment]
Its a nerf, but don't quite see it as a huge nerf? Radiation doesn't do that much damage unless you get a ton of it.
Max of 15 Tox per teleport, which would require really bad luck.
Average of about 3.75 Tox per teleport unless they teleport rapidly enough to hit 50 Radiation.

**Comment by tkdrg** [comment]
I guess, I overreacted a bit.

**Comment by GunHog** [comment]
I do not believe that this nerf is necessary.

**Comment by Boredone** [comment]
Personally, the main reasons for the nerf attempt is the fact that if the teleport triggers, the armor wearer takes no damage, and there's not much disadvantage to having it on if you arn't near space thanks to that.
Technically, I think its probably one of the most defensive non-specialized armors on the station cause of the first thing?

**Comment by kevinz000** [comment]
Uh
Radiation can make you collapse
Some one decided collapsing will give you max stamina damage
If you collapse in space it's GGNORE because space damages you which makes you permastunned.
Pretty big nerf.

**Comment by PKPenguin321** [comment]
big nerf but fair nerf

**Comment by Boredone** [comment]
Checking the code Kevinz (https://github.com/tgstation/-tg-station/blob/13b7de40e9d462de4905b0af67f892a77e70772d/code/modules/mob/living/carbon/human/species.dm) that can only happen if you go over 50 radiation, and even thats RNG unless you hit 100. So you would have to teleport FOUR times quickly to MAYBE collapse.
Unless someone teleports rapidly before the radiation wears off, this change will mostly just cause some RNG minor tox damage.

**Comment by KazeEspada** [comment]
Basically, this only cucks you if you were going to get cucked anyways. The armor is supposed to provide an escape chance, not make you invincible.

**Comment by tkdrg** [comment]
Alright okay let's give this a try

**Pull request merged**